Orange/widgets/visualize/owmosaic.py

```diff
@@ -400,7 +400,10 @@ def set_data(self, data):
         self.closeContext()
         self.data = data
         self.init_combos(self.data)
-        if self.data is None or not len(self.data):
+        if (self.data is None or
```

Comment:
Almost there :) Now radon is complaining that this function (which was not exactly a diamond before) looks worse.

I tend to agree, as the data checking part of code is growing larger and larger. I would recommend extraction the whole if-elif-else block into a method called something like "_get_discrete_data" that gets the data and returns data that consists only of discrete variables. The whole block would then turn into `self.discrete_data = self._get_discrete_data(data)`

+points if the new method has a docstring :)